pulp_rpm/tests/functional/api/test_retention_policy.py

```diff
@@ -0,0 +1,152 @@
+# coding=utf-8
+"""Tests for the retention policy feature of repositories."""
+
+from collections import defaultdict
+
+from pulp_smash import config
+from pulp_smash.pulp3.utils import (
+    gen_repo,
+    get_added_content_summary,
+    get_content,
+    get_content_summary,
+    get_removed_content,
+    delete_orphans,
+    modify_repo,
+)
+
+from pulp_rpm.tests.functional.constants import (
+    PULP_TYPE_ADVISORY,
+    PULP_TYPE_PACKAGE,
+    RPM_ADVISORY_COUNT,
+    RPM_ADVISORY_CONTENT_NAME,
+    RPM_ADVISORY_INCOMPLETE_PKG_LIST_URL,
+    RPM_ADVISORY_UPDATED_VERSION_URL,
+    RPM_ADVISORY_DIFFERENT_PKGLIST_URL,
+    RPM_ADVISORY_DIFFERENT_REPO_URL,
+    RPM_ADVISORY_NO_DATES,
+    RPM_ADVISORY_TEST_ID,
+    RPM_ADVISORY_TEST_REMOVE_COUNT,
+    RPM_ADVISORY_TEST_ADDED_COUNT,
+    RPM_EPEL_URL,
+    RPM_FIXTURE_SUMMARY,
+    RPM_INVALID_FIXTURE_URL,
+    RPM_KICKSTART_FIXTURE_SUMMARY,
+    RPM_KICKSTART_FIXTURE_URL,
+    RPM_MIRROR_LIST_GOOD_FIXTURE_URL,
+    RPM_MIRROR_LIST_BAD_FIXTURE_URL,
+    RPM_MODULAR_FIXTURE_SUMMARY,
+    RPM_MODULAR_FIXTURE_URL,
+    RPM_PACKAGE_CONTENT_NAME,
+    RPM_PACKAGE_COUNT,
+    RPM_RICH_WEAK_FIXTURE_URL,
+    RPM_ADVISORY_TEST_ID_NEW,
+    RPM_UPDATED_UPDATEINFO_FIXTURE_URL,
+    RPM_REFERENCES_UPDATEINFO_URL,
+    RPM_SIGNED_FIXTURE_URL,
+    RPM_UNSIGNED_FIXTURE_URL,
+    RPM_SHA512_FIXTURE_URL,
+    SRPM_UNSIGNED_FIXTURE_URL,
+    SRPM_UNSIGNED_FIXTURE_ADVISORY_COUNT,
+    SRPM_UNSIGNED_FIXTURE_PACKAGE_COUNT
+)
+from pulp_rpm.tests.functional.utils import (
+    gen_rpm_client,
+    gen_rpm_remote,
+    monitor_task,
+    progress_reports,
+)
+from pulp_rpm.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+from pulp_rpm.tests.functional.utils import PulpTestCase
+
+from pulpcore.client.pulp_rpm import (
+    RepositoriesRpmApi,
+    RpmRepositorySyncURL,
+    RemotesRpmApi,
+)
+
+
+class RetentionPolicyTestCase(PulpTestCase):
+    """Sync a repository with the rpm plugin."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        delete_orphans(cls.cfg)
+
+    def test_sync_with_retention(self):
```

Comment:
@goosemania This functional test only tests the retention policy feature with sync, do you think that is sufficient for now?

Reply:
Works for me. I saw you've already put a note in the coverage doc, thanks!